### PR TITLE
feat/JWT-3359_upgrate_to_gwt_2.9.0

### PR DESCRIPTION
**Pull request by benjaminalbanese**

### CODE REVIEW

**Comment by benjaminalbanese** [comment]
All tutorials are working so far, even the spring boot 001 tutorial which has now java 11 elements for demonstration purposes. Only the gatling tests should still be updated to be compatible to the new JWT version

Edit: Also, after the official 5.10.x.y release of the jadice web toolkit ist released, that version should be used instead of the 5.10.0.0. Only then i intend to merge the branch.

**Comment by benjaminalbanese** [comment]
With the last commit the gatling tests were fixed as well. The only remaining part is now, like written earlier, upgrading to the next stable jadice web toolkit version once that is officially published. 

p.s. : we might also want to update https://levigo.de/info/display/JKB/1+-+Szenario%3A+Grundlagen as there are 2 urls mentioned, that are hopelessly outdated and will be outdated again once the code is changed. Maybe add a warning mentioning this?

File: jwt-tutorial-spring-boot-001/pom.xml [hunk 1]
```diff
@@ -1,179 +1,193 @@
...
+  <dependencies>
+    <!-- Dependency versions are managed by parent / imported BOMs -->
+
+    <!-- JWT via Spring-Boot Starter -->
+    <dependency>
+      <groupId>com.levigo.jadice.webtoolkit</groupId>
+      <artifactId>webtoolkit-spring-boot-starter</artifactId>
+    </dependency>
+
+    <!-- The following two dependencies are not needed for this tutorial,	but were needed at a point in the real
```
Comment (hoffmadd) [inline jwt-tutorial-spring-boot-001/pom.xml]: Consider removing the dependencies then, if they are not needed. We probably do not want to confuse the Getting started user with stuff which is not needed.
However, maybe we can add a Knowledgebase Article for the case where this compatibility fix was used / needed.

File: jwt-tutorial-spring-boot-001/src/main/java/org/jwt/server/dataprovider/UrlDocumentDataProvider.java [hunk 2]
```diff
@@ -35,7 +35,8 @@ public UrlHandle createSourceHandle(UrlSource source) {
 
 	@Override
 	public void read(Reader reader, UrlSource source) throws JadiceException, IOException {
-		InputStream documentStream = getResourceStream(source.getUrl());
+		// use a java 10 feature: type inference
```
Comment (hoffmadd) [inline jwt-tutorial-spring-boot-001/src/main/java/org/jwt/server/dataprovider/UrlDocumentDataProvider.java:38]: Tschakaa ;-)

**Review by hoffmadd — CHANGES_REQUESTED** [review]

File: jwt-tutorial-spring-boot-001/pom.xml (hunk 1 above)
Comment (benjaminalbanese) [reply jwt-tutorial-spring-boot-001/pom.xml]: checked again and we don't need them here even after upgrading to java 11 in the dockerfile, so i removed them.

Comment (hoffmadd) [reply jwt-tutorial-spring-boot-001/pom.xml]: +1

**Comment by Crow08** [comment]
in `Application.gwt.xml` is 
`<!DOCTYPE module PUBLIC "-//Google Inc.//DTD Google Web Toolkit 2.8.0//EN" "http://www.gwtproject.org/doctype/2.8.2/gwt-module.dtd">` defined. Is there a 2.9 dtd?

**Comment by Crow08** [comment]
The normal tutorials work as intended but the Spring tutorial always returns 404 without throwing an error...
Is this a problem on my end only?

**Comment by benjaminalbanese** [comment]
> in `Application.gwt.xml` is
> `<!DOCTYPE module PUBLIC "-//Google Inc.//DTD Google Web Toolkit 2.8.0//EN" "http://www.gwtproject.org/doctype/2.8.2/gwt-module.dtd">` defined. Is there a 2.9 dtd?

I searched for a new definition some months ago after gwt 2.9.0 was released, but i did not find anything, including within the gwt source code.

**Comment by benjaminalbanese** [comment]
> The normal tutorials work as intended but the Spring tutorial always returns 404 without throwing an error...
> Is this a problem on my end only?

I built the whole project again with jdk 11 after upgrading the jadice web toolkit version, and in a second run i built the spring-boot-000 project again with jdk 8. ( the 001 project aims at showing java 11 compatibility, so you can't built it with jdk 8)

In all scenarios i didn't hava a problem, the documets loaded as intended. Could you please build the project again after getting the latest commit from some hours ago and check again if the error occurs?

**Review by Crow08 — APPROVED** [review]

**Review by hoffmadd — APPROVED** [review]

**Pull request merged**